### DISCUSSION TITLE
volume.check.disk for clusters with more than 1 replica

### DISCUSSION THREAD

**Discussion by tobiasmuehl** [seaweedfs/seaweedfs#2975] [Q&A]
Looking at the implementation of [`volume.check.disk`](https://github.com/chrislusf/seaweedfs/blob/master/weed/shell/command_volume_check_disk.go#L67) it looks like it is picking 2 replicas to compare to each other. Let's assume a volume is replicated 4 times. `volume.check.disk` runs once and compares/fixes 2 replicas. The other 2 replicas are untouched by the check and some writes might still be missing.

When running the command again, will the other 2 replicas be checked?

If [`collectVolumeReplicaLocations`](https://github.com/chrislusf/seaweedfs/blob/master/weed/shell/command_volume_fix_replication.go#L165:6) returns replicas in random order, I would assume the answer is that when running the command repeatedly, eventually all replicas will be in sync.

**Comment by chrislusf** [comment]
it sorts all replicas and check neighboring pairs https://github.com/chrislusf/seaweedfs/blob/master/weed/shell/command_volume_check_disk.go#L93
